packages/utils/src/lib/reports/generate-md-reports-diff.ts

```diff
@@ -0,0 +1,297 @@
+    return '';
+  }
+  return paragraphs(
+    h2('🎗️ Groups'),
+    formatGroupsOrAuditsDetails('group', diff.groups, {
+      headings: [
+        '🔌 Plugin',
+        '🎗️ Group',
+        '⭐ Current score',
+        '⭐ Previous score',
+        '🗠 Score change',
+      ],
+      rows: sortChanges(diff.groups.changed).map(group => [
+        group.plugin.title,
+        group.title,
+        formatScoreWithColor(group.scores.after),
+        formatScoreWithColor(group.scores.before, { skipBold: true }),
+        formatScoreChange(group.scores.diff),
+      ]),
+      align: ['l', 'l', 'c', 'c', 'c'],
+    }),
+  );
+}
+
+function formatDiffAuditsSection(diff: ReportsDiff): string {
+  return paragraphs(
+    h2('🛡️ Audits'),
+    formatGroupsOrAuditsDetails('audit', diff.audits, {
+      headings: [
+        '🔌 Plugin',
+        '🛡️ Audit',
+        '📏 Current value',
+        '📏 Previous value',
+        '🗠 Value change',
+      ],
+      rows: sortChanges(diff.audits.changed).map(audit => [
+        audit.plugin.title,
+        audit.title,
+        `${getSquaredScoreMarker(audit.scores.after)} ${style(
+          audit.displayValues.after || audit.values.after.toString(),
+        )}`,
+        `${getSquaredScoreMarker(audit.scores.before)} ${
+          audit.displayValues.before || audit.values.before.toString()
+        }`,
+        formatValueChange(audit),
+      ]),
+      align: ['l', 'l', 'c', 'c', 'c'],
+    }),
+  );
+}
+
+function formatGroupsOrAuditsDetails<T extends 'group' | 'audit'>(
+  token: T,
+  { changed, unchanged }: ReportsDiff[`${T}s`],
+  table: { headings: string[]; rows: string[][]; align?: Alignment[] },
+): string {
+  return changed.length === 0
+    ? summarizeUnchanged(token, { changed, unchanged })
+    : details(
+        summarizeDiffOutcomes(changesToDiffOutcomes(changed), token),
+        paragraphs(
+          tableMd(
+            [table.headings, ...table.rows.slice(0, MAX_ROWS)],
+            table.align,
+          ),
+          changed.length > MAX_ROWS &&
+            style(
+              `Only the ${MAX_ROWS} most affected ${pluralize(
+                token,
+              )} are listed above for brevity.`,
+              ['i'],
+            ),
+          unchanged.length > 0 &&
+            summarizeUnchanged(token, { changed, unchanged }),
+        ),
+      );
+}
+
+function formatScoreChange(diff: number): string {
+  const marker = getDiffMarker(diff);
+  const text = formatDiffNumber(Math.round(diff * 100));
+  return colorByScoreDiff(`${marker} ${text}`, diff);
+}
+
+function formatValueChange({
+  values,
+  scores,
+}: Pick<AuditDiff, 'values' | 'scores'>): string {
+  const marker = getDiffMarker(values.diff);
+  const percentage =
+    values.before === 0
+      ? values.diff > 0
+        ? Number.POSITIVE_INFINITY
+        : Number.NEGATIVE_INFINITY
+      : Math.round((100 * values.diff) / values.before);
+  const text = `${formatDiffNumber(percentage)}%`;
```

Comment:
Shouldn't there be a whitespace between a number and percentage?
```suggestion
  const text = `${formatDiffNumber(percentage)} %`;
```

Reply:
If you think it looks better, I can change it.

![image](https://github.com/code-pushup/cli/assets/34691111/45c8fc9b-8dd4-4cf1-b343-c208d6c7ad6e)

Reply:
Added thin space character.